processor/src/lib.rs

```diff
@@ -0,0 +1,171 @@
+use log::debug;
+use vm_core::program::Program;
+use crate::memory::Memory;
+
+
+#[cfg(test)]
+mod tests;
+mod memory;
+
+/// fixme: use 16 registers
+const REGISTER_NUM: usize = 16;
+
+#[derive(Debug, Default)]
+pub struct Process {
+    pub clk: u32,
+    // todo: use felt
+    pub registers: [u32; REGISTER_NUM],
+    pub pc: usize,
+    // todo: fp stack need be modified for procedure
+    pub fp: Vec<usize>,
+    pub flag: bool,
+    pub memory: Memory,
+}
+
+impl Process {
+    pub fn new() -> Self {
+        return Process{
+            clk: 0,
+            registers : [0; REGISTER_NUM],
+            pc: 0,
+            fp: Vec::new(),
+            flag: false,
+            memory: Default::default()
+        };
+    }
+
+    pub fn get_reg_index(&self, reg_str: &str) -> usize {
+        let first = reg_str.chars().nth(0).unwrap();
+        assert!(first ==  'r', "wrong reg name");
+        let index = reg_str[1..].parse().unwrap();
+        return index;
+    }
+
+    pub fn get_index_value(&self, op_str: &str)-> u32 {
+        let src = op_str.parse();
+        let mut value = Default::default();
+        if src.is_ok() {
+            value = src.unwrap();
+        } else {
+            let src_index = self.get_reg_index(op_str);
+            value = self.registers[src_index];
+        }
+        value
+    }
+
+    pub fn execute(&mut self, program: &Program) -> Result<(), String> {
+        loop {
+            let ops_line =  program.instructions[self.pc].trim();
+            let ops: Vec<&str> = ops_line.split(' ').collect();
+            let mut opcode = ops.get(0).unwrap().to_lowercase();
```

Comment:
compiled instruction structure might in a different form, should rewrite the parse logic.